stdlib/public/core/StringCharacterView.swift

```diff
@@ -445,7 +445,14 @@ extension String.CharacterView : BidirectionalCollection {
       let relativeOffset = i._base._position - _coreOffset
       if _core.isASCII {
         let asciiBuffer = _core.asciiBuffer._unsafelyUnwrappedUnchecked
-        return Character(UnicodeScalar(asciiBuffer[relativeOffset]))
+        // Bounds checks in an UnsafeBufferPointer (asciiBuffer) are only
+        // performed in Debug mode, so they need to be duplicated here.
+        // Falling back to the non-optimal behavior in the case they don't
+        // pass.
+        if relativeOffset >= asciiBuffer.startIndex &&
+          relativeOffset < asciiBuffer.endIndex {
+          return Character(UnicodeScalar(asciiBuffer[relativeOffset]))
```

Comment:
The check is performed somewhere outside the fastPath already, I'm just letting it do it's job. And it did not slow down the benchmarks. If it's going to trap anyway, is there any benefit in doing it quickly at the beginning?